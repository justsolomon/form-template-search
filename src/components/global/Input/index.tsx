import styles from './input.module.scss';

interface InputProps extends React.InputHTMLAttributes<HTMLInputElement> {
  leftIcon?: string | React.ReactNode;
  rightIcon?: string | React.ReactNode;
  containerClass?: string;
}

const Input = ({
  leftIcon,
  rightIcon,
  containerClass,
  ...rest
}: InputProps) => {
  return (
    <div className={`${styles['input-group']} ${containerClass}`}>
      {leftIcon ? (
        <div className={styles['input-group__left-element']}>{rightIcon}</div>
      ) : null}

      <input
        className={`${styles['input-group__input']} ${
          leftIcon ? styles['input-group__input--has-left'] : ''
        } ${rightIcon ? styles['input-group__input--has-right'] : ''}`}
        {...rest}
      />

      {rightIcon ? (
        <div className={styles['input-group__right-element']}>{rightIcon}</div>
      ) : null}
    </div>
  );
};

export default Input;
