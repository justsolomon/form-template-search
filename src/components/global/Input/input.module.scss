@import 'styles/variables';
@import 'styles/mixins';

.input-group {
  @include flex-center();

  width: 320px;
  position: relative;
  justify-content: flex-start;

  &__input {
    height: 45px;
    width: 100%;
    padding: 0 32px;
    border: 1px solid $input-border;
    border-radius: $border-radius--sm;
    background-color: $white;

    &::placeholder {
      color: $placeholder-color;
    }

    &--has-left {
      padding-left: 48px;
    }

    &--has-right {
      padding-right: 48px;
    }

    &:focus {
      outline: none;
    }
  }

  &__left-element,
  &__right-element {
    @include input-group-element();

    justify-content: center;
  }

  &__left-element {
    &--overflow {
      width: unset;
    }

    span {
      font-size: $font-size--md;
      margin: -3px 0 0 8px;
    }
  }

  &__right-element {
    left: unset;
    right: 0;
  }

  @media #{$media-mobile--sm} {
    &__input {
      padding: 0 16px;
    }
  }
}
