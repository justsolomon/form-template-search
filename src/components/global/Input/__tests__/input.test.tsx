import { render } from '@testing-library/react';
import Input from '..';

describe('<Input />', () => {
  it('should render the Input component correctly', () => {
    const { asFragment } = render(
      <Input defaultValue="Test input" placeholder="Test placeholder" />,
    );

    expect(asFragment).toMatchSnapshot();
  });
});
