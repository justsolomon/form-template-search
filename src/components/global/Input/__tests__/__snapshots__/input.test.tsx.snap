// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should render the Input component correctly 1`] = `[Function]`;
